### PR TITLE
General improvements to everything

### PR DESCRIPTION
**Pull request by zdenek-biberle**

### CODE REVIEW

File: source/app.d
Comment (Hnatekmar) [inline source/app.d]: Wouldn't ```(x - circle.x) ^^ 2 + (y - circle.y) ^^ 2 <= rad2 ^^ 2``` be more compact? Furthermore what was wrong with inlined distance?

File: source/app.d
Comment (Hnatekmar) [inline source/app.d]: Should be ```color.g``` instead of ```color.r``` :)

File: source/app.d
Comment (Hnatekmar) [inline source/app.d]: Same here change ```color.r``` to ```color.b```

File: source/app.d
Comment (Hnatekmar) [inline source/app.d:110]: Why is there ```+1```?

**Review by Hnatekmar — CHANGES_REQUESTED** [review]
Thank you for your PR. It is much obliged and it really looks like it is going to make application both faster and more readable (kudos for that). 
However I cannot approve it yet since there are some bugs that should be fixed before I merge this into master. Please fix them so I can merge this. I also left some questions about parts of code I did not fully understand if you could indulge me and comment on that it would be much appreciated :)
Other than that lgtm 👍

File: source/app.d
Comment (zdenek-biberle) [reply source/app.d]: Yup.

File: source/app.d
Comment (zdenek-biberle) [reply source/app.d]: Yup.

File: source/app.d
Comment (zdenek-biberle) [reply source/app.d]: It would and it actually produces the same code.

The problem with the original `distance` is that calculating a square root is expensive.

File: source/app.d
Comment (zdenek-biberle) [reply source/app.d:110]: Because an interval like left..right is right-open.

File: source/app.d
Comment (Hnatekmar) [reply source/app.d]: Thanks for clarification 👍

File: source/app.d
Comment (Hnatekmar) [reply source/app.d:110]: I see

**Review by Hnatekmar — APPROVED** [review]

**Pull request merged**
